### PR TITLE
[dep] Bump the default k8s version used to 1.27

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Bump the default k8s version used in integration and e2e tests to 1.27

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,13 +54,13 @@ BASE_IMAGE ?= gcr.io/distroless/static:nonroot
...
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
-ENVTEST_K8S_VERSION = 1.26
+ENVTEST_K8S_VERSION = 1.27
 
 INTEGRATION_TARGET ?= ./test/integration/...
 
 E2E_TARGET ?= ./test/e2e/...
 
-E2E_KIND_VERSION ?= kindest/node:v1.24.7
+E2E_KIND_VERSION ?= kindest/node:v1.27.2
```
[the remark this answers is not in the document]
Comment (tenzen-y) [reply Makefile:63]: @trasc Any reason that you want to use the latest version?

**Review by tenzen-y — COMMENTED** [review]
As of the scheduled release date of 0.4.0 (IIRC this month), 1.24 will not yet reach EoL.
So we should decide which k8s version will be supported by kueue 0.4.0 before updating the dependency version.

cc: @kerthcet @alculquicondor

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (trasc) [reply Makefile:63]: I'm fine reverting this. @alculquicondor?

File: Makefile [hunk 2]
```diff
@@ -54,13 +54,13 @@ BASE_IMAGE ?= gcr.io/distroless/static:nonroot
 BUILDER_IMAGE ?= golang:$(GO_VERSION)
 
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
-ENVTEST_K8S_VERSION = 1.26
+ENVTEST_K8S_VERSION = 1.27
```
Comment (kerthcet) [inline Makefile:57]: Maybe we should bump the k8s to 1.27 the same time.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (kerthcet) [reply Makefile:63]: +1 with @tenzen-y

File: Makefile (hunk 2 above)
Comment (trasc) [reply Makefile]: Maybe keep the default versions in sync with the deps (1.26)
https://github.com/kubernetes-sigs/kueue/blob/e8209f25f047f299929b9c349f4ee80f4d120ca5/go.mod#L16-L21

Comment (alculquicondor) [reply Makefile]: I think it's time to update the deps, rather :)

Comment (trasc) [reply Makefile]: We can hold this as is until we update the deps.

Comment (tenzen-y) [reply Makefile]: > I think it's time to update the deps, rather :)

+1

**Comment by alculquicondor** [comment]
This could be some potential flakiness:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/856/pull-kueue-test-e2e-main-1-26/1673348839160942592

/retest
as this PR should be unrleated.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (trasc) [reply Makefile]: done

File: Makefile [hunk 3]
```diff
@@ -54,13 +54,13 @@ BASE_IMAGE ?= gcr.io/distroless/static:nonroot
...
 
 INTEGRATION_TARGET ?= ./test/integration/...
 
 E2E_TARGET ?= ./test/e2e/...
 
-E2E_KIND_VERSION ?= kindest/node:v1.24.7
+E2E_KIND_VERSION ?= kindest/node:v1.27.2
 
 # E2E_K8S_VERSIONS sets the list of k8s versions included in test-e2e-all
 E2E_K8S_VERSIONS ?= 1.24.7 1.25.3 1.26.2 1.27.2
```
[the remark this answers is not in the document]
Comment (trasc) [reply Makefile]: done

File: go.mod [hunk 4]
```diff
@@ -9,21 +9,22 @@ require (
...
-	k8s.io/api v0.26.6
-	k8s.io/apiextensions-apiserver v0.26.6
-	k8s.io/apimachinery v0.26.6
-	k8s.io/apiserver v0.26.6
-	k8s.io/client-go v0.26.6
-	k8s.io/code-generator v0.26.6
-	k8s.io/component-base v0.26.6
-	k8s.io/component-helpers v0.26.6
+	k8s.io/api v0.27.3
+	k8s.io/apiextensions-apiserver v0.27.2
```
[the remark this answers is not in the document]
Comment (trasc) [reply go.mod]: done

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve
/hold cancel

**Pull request merged**